engine/lib/luajit-ffi-gen/src/impl_item/type_info.rs

```diff
@@ -62,22 +62,48 @@ const COPY_TYPES: &[&str] = &[
     "FrameStage",
 ];
 
+#[derive(Debug, PartialEq)]
+pub enum TypeWrapper {
+    /// Base type T, no wrapper.
+    None,
+    /// Option type: Option<T>, Option<&T>, Option<&mut T>
+    Option,
+    /// Boxed type: Box<T>
+    Box,
+    /// Slice type: [T]
```

Comment:
I suppose we are going to support `&[T]`/`&mut [T]` types only, since they include a size of the slice?

Reply:
Yeah, I don't think it's possible to have a slice without a reference operator. That being said, it would probably be nice to support fixed size arrays as well e.g. `[T; N]`